app/src/main/kotlin/com/wire/android/ui/authentication/login/Login.kt

```diff
@@ -32,25 +33,32 @@ import androidx.compose.ui.text.input.KeyboardType
 import androidx.compose.ui.text.input.TextFieldValue
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.text.style.TextDecoration
-import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.wire.android.R
+import com.wire.android.feature.auth.login.LoginViewModel
 import com.wire.android.ui.common.button.WireButtonState
 import com.wire.android.ui.common.textfield.WirePasswordTextField
 import com.wire.android.ui.common.textfield.WirePrimaryButton
 import com.wire.android.ui.common.textfield.WireTextField
 import com.wire.android.ui.theme.body02
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 
 
-@Preview
+//@Preview
 @Composable
 fun LoginScreen() {
-    LoginContent()
+    val scope = rememberCoroutineScope()
+    LoginContent(scope = scope)
 }
 
 @OptIn(ExperimentalComposeUiApi::class, ExperimentalMaterial3Api::class)
 @Composable
-private fun LoginContent() {
+private fun LoginContent(
+    loginViewModel: LoginViewModel = hiltViewModel(),
+    scope: CoroutineScope
+) {
 
     var email by remember { mutableStateOf(TextFieldValue("")) }
     var password by remember { mutableStateOf(TextFieldValue("")) }
```

Comment:
I am not an expert but in MVI it should be working like this : 

UI > Send Event to ViewModel > ViewModel process the event > ViewModel emits new state of the State object 

so I would move this "email" and "password" fields into the ViewModel and react to it, do some validation etc.